api/src/main/java/org/geysermc/geyser/api/pack/ResourcePack.java

```diff
@@ -59,6 +62,25 @@ public interface ResourcePack {
     @NonNull
     String contentKey();
 
+    /**
+     * Sets the content key of the resource pack. Lack of a content key can be represented by an empty string.
+     */
+    void contentKey(@Nullable String contentKey);
+
+    /**
+     * The subpack to tell Bedrock clients to load. Lack of a subpack to load is represented by an empty string.
+     *
+     * @return the subpack name, or an empty string if not set.
+     */
+    @NonNull
+    String subpackName();
+
+    /**
+     * Sets the subpack name that clients should load.
+     * It must match one of the subpacks that can be found in the manifest.
+     */
+    void subpackName(@Nullable String subpackName);
```

Comment:
The addition of the two mutations seems dangerous. The first thought is that we don't want to be mutating packs that are in the Geyser registry just because of the way we are sending them to the client.

More importantly, packs need to be thread safe because they are shared by sessions. 

We could add additions to SessionLoadResourcePacksEvent, without need for deprecations:
```java
public abstract @NonNull List<Entry> entries();
public abstract boolean register(@NonNull ResourcePack resourcePack, @NonNull String subpack);

public interface Entry {
    @NonNull ResourcePack resourcePack();
    @NonNull String subpack();
}
```

Reply:
Personally, I'd like to avoid adding multiple register methods - they'd have to be duplicated for the eventual GeyserDefineResourcePacks event

Reply:
Then make `Entry` a separate class and use that for any register methods that take any more parameters than just `ResourcePack`. Or something like that.